OpenSim/Tests/Wrapping/testWrapCylinder.cpp

```diff
@@ -0,0 +1,314 @@
+    struct WrapTestResult final {
+
+        static WrapTestResult NoWrap() {
+            WrapTestResult result;
+            result.noWrap = true;
+            return result;
+        }
+
+        // Path segment on cylinder surface.
+        PathSegment path = {{}, {}};
+        // Path segment length.
+        double length = NAN;
+        // True if there is no wrapping (the other fields don't matter).
+        bool noWrap = false;
+    };
+
+    std::ostream& operator<<(
+        std::ostream& os,
+        const WrapTestResult& result)
+    {
+        if (result.noWrap) {
+            return os << "WrapTestResult{noWrap = true}";
+        }
+        return os <<
+            "WrapTestResult{" <<
+            "path: " << result.path << ", " <<
+            "length: " << result.length << "}";
+    }
+
+    // Struct holding the tolerances when asserting the wrapping result.
+    struct WrappingTolerances final {
+        WrappingTolerances() = default;
+
+        WrappingTolerances(double eps):
+            position(eps),
+            length(eps)
+        {}
+
+        double position = 1e-9;
+        double length = 1e-9;
+    };
+
+    bool IsEqual(
+        double lhs,
+        double rhs,
+        double tolerance)
+    {
+        return std::abs(lhs - rhs) < tolerance;
+    }
+
+    bool IsEqual(
+        const SimTK::Vec3& lhs,
+        const SimTK::Vec3& rhs,
+        double tolerance)
+    {
+        return IsEqual(lhs[0], rhs[0], tolerance)
+            && IsEqual(lhs[1], rhs[1], tolerance)
+            && IsEqual(lhs[2], rhs[2], tolerance);
+    }
+
+    bool IsEqual(
+        const PathSegment& lhs,
+        const PathSegment& rhs,
+        double tolerance)
+    {
+        return IsEqual(lhs.start, rhs.start, tolerance)
+            && IsEqual(lhs.end, rhs.end, tolerance);
+    }
+
+    bool IsEqual(
+        const WrapTestResult& lhs,
+        const WrapTestResult& rhs,
+        const WrappingTolerances& tolerance)
+    {
+        if (lhs.noWrap && rhs.noWrap) {
+            return true;
+        }
+        return IsEqual(lhs.path, rhs.path, tolerance.position)
+            && IsEqual(lhs.length, rhs.length, tolerance.length)
+            && lhs.noWrap == rhs.noWrap;
+    }
+
+}
+
+// Section on configuring and simulating a specific wrapping scenario.
+namespace {
+
+    // Parameterization of the scenario for testing the wrapping.
+    //
+    // The simulated scene is one with a cylinder at the origin, with a user
+    // defined radius, length, active quadrant, and relative orientation.
+    // A specific wrapping case can be configured by choosing the start and end
+    // points of the path.
+    struct WrapInput final {
+
+        SimTK::Rotation cylinderOrientation() const {
+            SimTK::Rotation orientation;
+            orientation.setRotationToBodyFixedXYZ(eulerRotations);
+            return orientation;
+        }
+
+        // Endpoints of the total path:
+        PathSegment path = {{}, {}};
```

Comment:
`SimTK::Vec3`'s initialization rules imply that this defaults to "junk" in release mode (I believe?)

Reply:
Yeah looks like that is the case